server/player/player.go

```diff
@@ -964,6 +989,22 @@ func (p *Player) Respawn() {
 	p.SetVisible()
 }
 
+// realSpawnPos returns position and world where player should be spawned.
+func (p *Player) realSpawnPos() (pos cube.Pos, w *world.World, spawnBlockBroken bool, previousDimension world.Dimension) {
+	w = p.World()
+
+	previousDimension = w.Dimension()
+	pos = w.PlayerSpawn(p.UUID())
+	if b, ok := w.Block(pos).(block.RespawnBlock); ok && b.CanSpawn() {
+		return pos, w, false, previousDimension
+	}
+
+	// We can use the principle here that returning through a portal of a specific dimension inside that dimension will
+	// always bring us back to the overworld.
+	w = w.PortalDestination(w.Dimension())
+	return w.Spawn(), w, true, previousDimension
+}
```

Comment:
The name and documentation for this method is slightly misleading since nowhere does it state it actually calls w.Spawn()

Reply:
ok, I'll call this function ¨spawnLocation¨